### PR TITLE
First draft of porting mageck count to nf-test

### PR DESCRIPTION
**Pull request by LaurenceKuhl**
## PR checklist

Closes #XXX 

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [ ] Emit the `versions.yml` file.
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - For modules:
    - [ ] `nf-core modules test <MODULE> --profile docker`
    - [ ] `nf-core modules test <MODULE> --profile singularity`
    - [ ] `nf-core modules test <MODULE> --profile conda`
  - For subworkflows:
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile docker`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile singularity`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile conda`

### CODE REVIEW

File: modules/nf-core/mageck/count/main.nf [hunk 1]
```diff
@@ -35,6 +35,18 @@ process MAGECK_COUNT {
...
+        mageck: \$(mageck -v)
+    END_VERSIONS
+    """
+    stub:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    def input_file = ("$inputfile".endsWith(".fastq.gz")) ? "--fastq ${inputfile}" : "-k ${inputfile}"
+    def sample_label = ("$inputfile".endsWith(".fastq.gz") || "$inputfile".endsWith(".fq.gz")) ? "--sample-label ${meta.id}" : ''
+    """
+    touch count_table.count_normalized.txt
```
Comment (famosab) [inline modules/nf-core/mageck/count/main.nf:49]:
```suggestion
    touch ${prefix}.count_normalized.txt
```

File: modules/nf-core/mageck/count/tests/main.nf.test [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    tag "mageck/count"
+
+    test("mageck-count - fastq") {
+
+        config "./nextflow.config"
+
+        when {
+            process {
+                """
+                input[0] = [ [ id:'test,test2', single_end:true] , // meta map
```
Comment (famosab) [inline modules/nf-core/mageck/count/tests/main.nf.test:21]: You might want to adjust the filepaths to look like this:
`file(params.modules_testdata_base_path + 'genomics/mus_musculus/mageck/count_table.csv', checkIfExists: true)`
But lets solve the other issues first

File: modules/nf-core/mageck/count/tests/main.nf.test [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+                """
+            }
+        }
+
+        then {
+            assertAll(
+                { assert process.success },
+                { assert snapshot(
+                            process.out.versions,
+                            file(process.out.count.get(0).get(1)).readLines()[0]).match() }
```
Comment (famosab) [inline modules/nf-core/mageck/count/tests/main.nf.test:35]:
```suggestion
                            file(process.out.count_norm.get(0).get(1)).readLines()[0]).match() }
```

Comment (famosab) [reply modules/nf-core/mageck/count/tests/main.nf.test:35]: or can you check both?

File: modules/nf-core/mageck/count/tests/main.nf.test.snap
Comment (famosab) [inline modules/nf-core/mageck/count/tests/main.nf.test.snap:1]: This file looks wrong, there are no entries anywhere 🤔 

`nf-core modules test mageck/count -profile docker -u`

File: modules/nf-core/mageck/count/tests/main.nf.test (hunk 3 above)
Comment (LaurenceKuhl) [reply modules/nf-core/mageck/count/tests/main.nf.test:35]: there's norm and count but not count_norm, i can add norm for sure as soon as the tests run

Comment (famosab) [reply modules/nf-core/mageck/count/tests/main.nf.test]: yes I realized, sorry! I changed it in my commit

File: modules/nf-core/mageck/count/main.nf [hunk 4]
```diff
@@ -35,6 +35,18 @@ process MAGECK_COUNT {
...
+        mageck: \$(mageck -v)
+    END_VERSIONS
+    """
+    stub:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    def input_file = ("$inputfile".endsWith(".fastq.gz")) ? "--fastq ${inputfile}" : "-k ${inputfile}"
+    def sample_label = ("$inputfile".endsWith(".fastq.gz") || "$inputfile".endsWith(".fq.gz")) ? "--sample-label ${meta.id}" : ''
+    """
+    touch ${prefix}.count_normalized.txt
```
Comment (famosab) [inline modules/nf-core/mageck/count/main.nf:49]: the stub is not complete: we also need to create the file emitted by count

File: modules/nf-core/mageck/count/tests/main.nf.test [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+    name "Test Process MAGECK_COUNT"
+    script "../main.nf"
+    process "MAGECK_COUNT"
+
+    tag "modules"
+    tag "modules_nfcore"
+    tag "mageck"
+    tag "mageck/count"
+
+    test("mus_musculus - fastq") {
```
Comment (famosab) [inline modules/nf-core/mageck/count/tests/main.nf.test:13]: ah and we need to name the tests by input data, i thought it was mouse data but maybe its something else

File: modules/nf-core/mageck/count/tests/main.nf.test [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+
+       // config "./nextflow.config"
+
+        when {
+            process {
+                """
+                input[0] = [ [ id:'test,test2', single_end:true] , // meta map
+                [file(params.modules_testdata_base_path + 'crisprseq/testdata/SRR23101364_1.small.fastq.gz', checkIfExists: true),
```
Comment (famosab) [inline modules/nf-core/mageck/count/tests/main.nf.test:21]: Now I get this:
  ERROR ~ No such file or directory:                                         │
│ https://raw.githubusercontent.com/nf-core/test-datasets/modules/data/crisprs │
│ eq/testdata/SRR23101364_1.small.fastq.gz

Maybe
```suggestion
                [file(params.modules_testdata_base_path + 'genomics/crisprseq/testdata/SRR23101364_1.small.fastq.gz', checkIfExists: true),
```
??

Comment (famosab) [reply modules/nf-core/mageck/count/tests/main.nf.test]: ah nvm we need to add the files to the repo first 
https://github.com/nf-core/test-datasets/blob/modules/docs/ADD_NEW_DATA.md

File: modules/nf-core/mageck/count/tests/main.nf.test [hunk 7]
```diff
@@ -0,0 +1,61 @@
+// nf-core modules test mageck/count
```
Comment (famosab) [inline modules/nf-core/mageck/count/tests/main.nf.test:1]:
```suggestion
```

**Review by famosab — APPROVED** [review]
LGTM!

**Pull request merged**